st2client/st2client/commands/action.py

```diff
@@ -1055,6 +1055,11 @@ def __init__(self, resource, *args, **kwargs):
                                       'and optional args for the action.')
         self.parser.add_argument('--tasks', nargs='*',
                                  help='Name of the workflow tasks to re-run.')
+        self.parser.add_argument('--no-reset', dest='no_reset', nargs='*',
```

Comment:
I'm willing to experiment here. The current option here take a list of task names to not reset. This is to accommodate use cases where there are multiple with-items tasks and user don't want to reset a subset of them.  A global flag would limit it.